token/services/ttx/endorse.go

```diff
@@ -736,8 +738,8 @@ func (f *ReceiveTransactionView) Call(context view.Context) (interface{}, error)
 	span := trace.SpanFromContext(context.Context())
 	span.AddEvent("start_receive_transaction_view")
 	defer span.AddEvent("end_receive_transaction_view")
-
-	msg, err := ReadMessage(context.Session(), time.Minute*4)
+	jsonSession := session2.JSON(context)
+	msg, err := jsonSession.ReceiveRawWithTimeout(time.Minute * 4)
```

Comment:
we could also extract these durations to constants, depending on what we are waiting for, e.g. `signatureRcvTimeout`, `txRcvTimeout`, to make sure we are aligned and to have a better overview. ideally I would even add them as fields in the structure objects and set the default values in the constructors.